docs/feature_pointing_device.md

```diff
@@ -822,3 +825,694 @@ In general the following two functions must be implemented in appropriate locati
+
+Speed and sensitivity of any mode will be impacted by the pointing device CPI setting so divisors may need to be adjusted to personal preference and CPI settings typically used (_note dynamic divisors adjustment based on cpi could be used if desired_). 
+
+!> Drag scroll speed will be effected by OS mouse settings (_there are usually separate settings for scroll "wheel" and "wheel tilt" which is Vertical and Horzontal scroll respectively_)   
+
+The `POINTING_DRAG_DIVISOR` default value of 16 is based on having mouse settings in the OS set to three lines per tick of "mouse wheel" or "wheel tilt" (_Windows Default_)
+
+!> `POINTING_PRECISION_DIVISOR` default will half cursor speed when active (_divisor of 2_) but a divisor of 4 is fine to use as well but the cursor will be quite a bit slower, however divisors of 8 or greater will likely only work well for very high cpi settings.
+
+Recommended settings for `POINTING_CARET_DIVISOR_V` and `POINTING_CARET_DIVISOR_H` are `16` and `32` respectively which will give preference to horizontal caret movement (`KC_LEFT`, `KC_RIGHT`) over vertical (`KC_UP`, `KC_DOWN`) giving even more stability to horizontal movement (_reducing errant up and down movement_).
+
+!> `POINTING_TAP_DELAY` defaults to `TAP_CODE_DELAY` so if `TAP_CODE_DELAY` is set high it can cause some noticable latency on some modes such as `PM_CARET`
+
+### Basic Custom Modes
+
+There are a couple of ways to add new pointing device modes, using the pointing device mode maps will be covered here under basic use where the other method of using the `process_pointing_mode_*` callbacks will be covered under advance use.
+
+#### Pointing Device Mode Maps
+
+The easiest way to add pointing device modes that are only using keycode taps (similar to `PM_CARET`, `PM_VOLUME`, and `PM_HISTORY`) are through creating pointing device mode maps.
+
+!> Currently only **basic keycodes with modifiers** are supported by pointing device mode maps and the `pointing_tap_codes` function (custom and quantum keycodes are unsupported at this time)
+
+```c
+// Pointing Device Mode Maps Format
+const uint16_t PROGMEM pointing_device_mode_maps[][POINTING_NUM_DIRECTIONS] = {
+    [0] = POINTING_MODE_LAYOUT(
+                <keycode up>,
+        <keycode left>,       <keycode right>,
+                <keycode down>
+    ),
+    //... all other pointing mode maps ...
+    [<POINTING_MODE_MAP_COUNT - 1>] = POINTING_MODE_LAYOUT(
+                <keycode up>,
+        <keycode left>,       <keycode right>,
+                <keycode down>
+    )
+}
+
+```
+
+#### Example Mode Maps:
+
+```c
+// in config.h:
+#define POINTING_DEVICE_MODES_ENABLE // (Required)
+#define POINTING_MODE_MAP_COUNT 4    // (Required) number of modes in map
```

Comment:
If this is required, it would be worth adding something like this to the header file: 

```c
#ifndef POINTING_MODE_MAP_COUNT
#    define POINTING_MODE_MAP_COUNT 0
#    error POINTING_MODE_MAP_COUNT is not defined.
#endif
```

This way, it's a lot more obvious what screwed up, and doesn't error all over the place.

Reply:
This is due to out of date documentation was going to update once the bugs are sorted but yes this should error if manually defined is now automatically defined at compile time.